newsletter/admin.py

```diff
@@ -43,13 +44,22 @@
 
 from .settings import newsletter_settings
 
+
+logger = logging.getLogger(__name__)
+
+
 # Contsruct URL's for icons
-ICON_URLS = {
-    'yes': '%snewsletter/admin/img/icon-yes.gif' % settings.STATIC_URL,
-    'wait': '%snewsletter/admin/img/waiting.gif' % settings.STATIC_URL,
-    'submit': '%snewsletter/admin/img/submitting.gif' % settings.STATIC_URL,
-    'no': '%snewsletter/admin/img/icon-no.gif' % settings.STATIC_URL
-}
+def icon_url(icon, alt_text, _urls={}):
+    if not _urls:
+        _urls['yes'] = static('newsletter/admin/img/icon-yes.gif')
+        _urls['wait'] = static('newsletter/admin/img/waiting.gif')
+        _urls['submit'] = static('newsletter/admin/img/submitting.gif')
+        _urls['no'] = static('newsletter/admin/img/icon-no.gif')
+
+    return format_html(
+        '<img src="{}" width="10" height="10" alt="{}"/>',
+        _urls[icon], alt_text
+    )
```

Comment:
Thanks for the comment on the SVG. Are you sure the HTML to render them is the same? I have a vague memory of the dimensions being set through CSS or something.

As with the URL's as cache: as it's currently unused I prefer to not have it there until it is. Moreover, I wonder if the speed gain is actually worth the additional code complexity: the admin is hardly ever the bottleneck and, generally, `static()` is called wherever URL's are generated from within admin templates. Actually, I think it isn't worth it - in which case the docstring of the function would be a lot simpler too. ;)